src/components/Company.js

```diff
@@ -5,11 +5,11 @@ import { useSelector } from 'react-redux';
 
 export const Company = ({ stock }) => {
     const state = useSelector((state) => state)
-    console.log(stock)
     return (
         <div className="company-overview">
             <h1 className="title">Company Overview</h1>
-            {stock.length === 0 ? <div className="loading-spinner"></div> :
+            {(stock === null) ? <div className="loading-spinner"></div> :
+            (stock.length === 0) ? <div> N/A </div> :
```

Comment:
Q: `stock` is an object, no? Does `stock.length` exist as a property on the object, or is this an error in thinking of it as an array?

Reply:
stock is an array for this company component, but we want to handle the initial state of null, what is the best way to do it?

Reply:
But Line 14 is using `stock.companyName`, `stock.symbol`, and Line 15 has `stock.website`, which all indicate it is an object and not an array.

Your handling of `null` is fine. I have a problem with the `stock.length === 0` check, as that will *always* be false (because `length` isn't a property of the object)

Reply:
Is this the correct way to do it?
`(Object.keys(stock).length === 0)`